### PR TITLE
[IMP] mail: remove propsCompareDepth from useShouldUpdateBasedOnProps

### PR DESCRIPTION
**Pull request by alexkuhn**
In preparation for using OWL v2 in discuss code.
(`shouldUpdateBasedOnProps` will be removed)

Task-2695743

### CODE REVIEW

**Comment by seb-odoo** [comment]
You'll need to rebase/fix conflicts.

**Comment by seb-odoo** [comment]
This PR has composer in title but it looks like it is for discuss mobile?
Am I missing something?

**Comment by alexkuhn** [comment]
> This PR has composer in title but it looks like it is for discuss mobile?
> Am I missing something?

It was the title for the 1st commit of the task, which has already been merged.
There are 4 commits with this task in total. 1st & 2nd already merged. 3rd is pending review/changes. 4th is final touch by removing `propsCompareDepth`.

I just changed the title so it's on the 4th commit.

File: addons/mail/static/src/component_hooks/use_should_update_based_on_props/use_should_update_based_on_props.js [hunk 1]
```diff
@@ -5,41 +5,21 @@ const { Component } = owl;
 const { onPatched } = owl.hooks;
 
 /**
- * Compares `a` and `b` up to the given `propsCompareDepth`.
+ * Shallow compares `a` and `b`.
  *
  * @param {any} a
  * @param {any} b
```
Comment (seb-odoo) [inline addons/mail/static/src/component_hooks/use_should_update_based_on_props/use_should_update_based_on_props.js:11]: a and b are now always props, so this doc could become Object instead (even before this PR I think it was really only Object or Array and never anything else)

File: addons/mail/static/src/component_hooks/use_should_update_based_on_props/use_should_update_based_on_props.js [hunk 2]
```diff
@@ -51,10 +31,8 @@ function isEqual(a, b, propsCompareDepth) {
  * comparing props.
  *
  * @param {Object} [param0={}]
```
Comment (seb-odoo) [inline addons/mail/static/src/component_hooks/use_should_update_based_on_props/use_should_update_based_on_props.js:33]: obsolete doc

File: addons/mail/static/src/utils/messaging_component.js [hunk 3]
```diff
@@ -17,9 +17,8 @@ const componentRegistry = {};
  * @param {Component} ComponentClass the constructor of the component to be
  *      registered. Its name will be used as its key in the registry.
  * @param {Object} [param1]
```
Comment (seb-odoo) [inline addons/mail/static/src/utils/messaging_component.js:19]: Still have obsolete doc here

**Review by seb-odoo — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**